test/e2e/fleet_test.go

```diff
@@ -222,123 +222,193 @@ func TestFleetScaleUpEditAndScaleDown(t *testing.T) {
-				assert.Equal(t, maxSurgeParam, flt.Spec.Strategy.RollingUpdate.MaxUnavailable.StrVal)
+			flt := defaultFleet(framework.Namespace)
+			flt.ApplyDefaults()
+			flt.Spec.Replicas = 1
+			rollingUpdatePercent := intstr.FromString(maxSurgeParam)
+			flt.Spec.Strategy.RollingUpdate.MaxSurge = &rollingUpdatePercent
+			flt.Spec.Strategy.RollingUpdate.MaxUnavailable = &rollingUpdatePercent
 
-				framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+			flt, err := client.Fleets(framework.Namespace).Create(ctx, flt, metav1.CreateOptions{})
+			if assert.Nil(t, err) {
+				defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+			}
 
-				// scale up
-				const targetScale = 8
-				if usePatch {
-					flt = scaleFleetPatch(ctx, t, flt, targetScale)
-					assert.Equal(t, int32(targetScale), flt.Spec.Replicas)
-				} else {
-					flt = scaleFleetSubresource(ctx, t, flt, targetScale)
-				}
+			assert.Equal(t, int32(1), flt.Spec.Replicas)
+			assert.Equal(t, maxSurgeParam, flt.Spec.Strategy.RollingUpdate.MaxSurge.StrVal)
+			assert.Equal(t, maxSurgeParam, flt.Spec.Strategy.RollingUpdate.MaxUnavailable.StrVal)
 
-				framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(targetScale))
+			framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
 
-				flt, err = client.Fleets(framework.Namespace).Get(ctx, flt.ObjectMeta.GetName(), metav1.GetOptions{})
-				assert.NoError(t, err)
+			// scale up
+			const targetScale = 8
+			if usePatch {
+				flt = scaleFleetPatch(ctx, t, flt, targetScale)
+				assert.Equal(t, int32(targetScale), flt.Spec.Replicas)
+			} else {
+				flt = scaleFleetSubresource(ctx, t, flt, targetScale)
+			}
 
-				// Change ContainerPort to trigger creating a new GSSet
+			framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(targetScale))
+
+			flt, err = client.Fleets(framework.Namespace).Get(ctx, flt.ObjectMeta.GetName(), metav1.GetOptions{})
+			assert.NoError(t, err)
+
+			done := make(chan bool, 1)
+			defer close(done)
+			if cycleAllocations {
+				// Repeatedly cycle allocations to keep ~half of the GameServers Allocated. Repeatedly Allocate and
+				// delete such that both the old and new GSSet contain allocated GameServers.
+				const halfScale = targetScale / 2
+				const period = 3 * time.Second
+				go framework.CycleAllocations(t, flt, period, period*halfScale, done)
+
+				// Wait for at least half of the fleet to have be cycled (either Allocated or shutting down)
+				// before updating the fleet.
+				err = framework.WaitForFleetCondition(t, flt, func(entry *logrus.Entry, fleet *agonesv1.Fleet) bool {
+					return fleet.Status.ReadyReplicas < halfScale
+				})
+			}
+
+			// Change ContainerPort to trigger creating a new GSSet
+			fltName := flt.GetName()
+			err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
So if I'm reading the DefaultBackoff correctly, it only retries 4 times, with only a 10ms gap-- which in a cluster under load, may very much not be enough.

My suggestion would be to either:

Create a backoff of something like:
```go
backoff := wait.Backoff {
   Duration: time.Second
   Factor: 1
   Cap: time.Minute
}
```

or it may be far simpler to use `require.Eventually(...)` with a 1 second rate, and a timeout of a minute.
https://pkg.go.dev/github.com/stretchr/testify/require#Eventually - since we end up in the same place anyway.

Reply:
so DefaultBackoff should have some exponential backoff (each time the delay is increased x5), if I read it correctly that will mean a max backoff of a couple of seconds. But not sure if it's catching the conflicts properly here, it's strange that it would still fail

I'll change it to `require.Eventually()` though, also to align with other tests